controller/api/apis.go

```diff
@@ -3460,3 +3460,53 @@ type RESTSelfApikeyData struct {
 	GlobalPermits []*RESTRolePermission            `json:"global_permissions,omitempty"`
 	DomainPermits map[string][]*RESTRolePermission `json:"domain_permissions,omitempty"` // domain -> permissions
 }
+
+type REST_SigstoreRootOfTrust_GET struct {
+	Name           string                           `json:"name"`
+	IsPrivate      bool                             `json:"is_private"`
+	RekorPublicKey string                           `json:"rekor_public_key,omitempty"`
+	RootCert       string                           `json:"root_cert,omitempty"`
+	SCTPublicKey   string                           `json:"sct_public_key,omitempty"`
+	Verifiers      map[string]REST_SigstoreVerifier `json:"verifiers,omitempty"`
+	CfgType        string                           `json:"cfg_type"`
+	Comment        string                           `json:"comment,omitempty"`
+}
+
+type REST_SigstoreRootOfTrust_POST struct {
+	Name           string `json:"name"`
+	IsPrivate      bool   `json:"is_private"`
+	RekorPublicKey string `json:"rekor_public_key,omitempty"`
+	RootCert       string `json:"root_cert,omitempty"`
+	SCTPublicKey   string `json:"sct_public_key,omitempty"`
+	CfgType        string `json:"cfg_type"`
+	Comment        string `json:"comment,omitempty"`
+}
+
+type REST_SigstoreRootOfTrust_PATCH struct {
+	Name           *string `json:"name"`
```

Comment:
name should be unchangeable & it should match the ":root_of_trust" in the url

Reply:
Since name is unchangeable, then it can be removed from this type, as the targeting will be done by the URL parameter.

Reply:
I think name should be in the object unless we are sure this REST object is only referenced in the api handlers.
If it's referenced somewhere else, it's hard to know what's the name from the object itself